update_fingerprints.py

```diff
@@ -133,8 +149,11 @@ def main(turns, repetitions):
 fp.plot()
 ```
     """.format(version, turns, repetitions)
+    try:
```

Comment:
Just a style comment -- I'd recommend splitting the read and create functions, or have the read function create the file if it doesn't exist. It's better IMO to have a function that creates a file if it doesn't exist and a separate function that reads.